### DISCUSSION TITLE
NOTE GRID CUSTOM FONT DESIGN

### DISCUSSION THREAD

**Discussion by worker-gray** [SynthstromAudible/DelugeFirmware#2621] [Ideas]
The idea is that you could customize your own fonts using the note entry grid (with the OLED pixel limitations)
It could be a mode you enter in the edit menu.
For character mapping, you select the character you want to edit via the qwerty layout on the note grid.  In this mode you will be able to jump in and out of the home screen. Home screen will be the QWERTY keyboard view. Unwritten characters will be grayed out, characters that have been altered will appear green. If shift+ character (in this mode) will call-up the character design screen with the font baseline indicated with a red mute.
Character width determined with <> and height determined with 2nd mute button. The height of the tallest characters will carry-over to all the other characters for universal row height, but the width will be handled on a per-character basis for optimized kerning (space between letters in a word) determined by 3 preset blank pixel selections-- small [1px], medium [2px], large [3px]. 4 pixel leading (space between lines of type) would suffice for descenders (like the tail of a lower case 'y')
You can save the font map during or after each character is created.
Font will not load unless all characters are mapped.
This might require a new folder (or a number of them)

**Comment by seangoodvibes** [comment]
An intriguing suggestion. I honestly don't know if it makes sense as part of the community firmware codebase (instead of a web based app), but it would be a fun project nonetheless. 

I just rendered a 8px by 8px icon from the grid:

https://github.com/user-attachments/assets/d5e312bf-982a-4885-b17b-8dceb5df114e

https://github.com/user-attachments/assets/8638a015-ef86-4302-8228-f23c70ca96f1

**Reply by worker-gray** [reply]
HOLY CRAP! I made a song on the deluge called that! (URL) it's on my soundcloud page--- but that's EXACTLY what I'm talking about! Even if it's web-based. A way to really customize the deluge (in a non-volatile way)

**Reply by worker-gray** [reply]
and dude-- you are super-talented. you make ideas come to life! thanks!!
